### ISSUE TITLE
test_timeout_expired is failing

### ISSUE THREAD

**Issue by PureTryOut** [mer-hybris/libgbinder#71]
```
mkdir -p build/debug
gcc -c -fPIC -Os -fomit-frame-pointer  -Wall -Wno-deprecated-declarations -I../common -I../../src -I../../include -MMD -MP -I/usr/include/gutil -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include  -g -DDEBUG -MT"build/debug/common_test_binder.o" -MF"build/debug/common_test_binder.d" ../common/test_binder.c -o build/debug/common_test_binder.o
gcc -c -fPIC -Os -fomit-frame-pointer  -Wall -Wno-deprecated-declarations -I../common -I../../src -I../../include -MMD -MP -I/usr/include/gutil -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include  -g -DDEBUG -MT"build/debug/common_test_main.o" -MF"build/debug/common_test_main.d" ../common/test_main.c -o build/debug/common_test_main.o
gcc -c -fPIC -Os -fomit-frame-pointer  -Wall -Wno-deprecated-declarations -I../common -I../../src -I../../include -MMD -MP -I/usr/include/gutil -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include  -g -DDEBUG -MT"build/debug/unit_ipc.o" -MF"build/debug/unit_ipc.d" unit_ipc.c -o build/debug/unit_ipc.o
make[2]: Entering directory '/home/bart/Documents/Git/alpine/aports/testing/libgbinder/src/libgbinder-1.1.10'
make[2]: Nothing to be done for 'debug_lib'.
make[2]: Leaving directory '/home/bart/Documents/Git/alpine/aports/testing/libgbinder/src/libgbinder-1.1.10'
gcc -fPIC -Wl,--as-needed -g build/debug/common_test_binder.o build/debug/common_test_main.o build/debug/unit_ipc.o ../../build/debug/libgbinder.a -lglibutil -lgobject-2.0 -lglib-2.0 -lintl  -lpthread -o build/debug/unit_ipc
# random seed: R02S4a10238b33f2a5a94d4ea56540420f67
1..23
# Start of ipc tests
ok 1 /ipc/null
ok 2 /ipc/basic
ok 3 /ipc/async_oneway
ok 4 /ipc/sync_oneway
ok 5 /ipc/sync_reply_ok
ok 6 /ipc/sync_reply_error
ok 7 /ipc/transact_ok
ok 8 /ipc/transact_dead
ok 9 /ipc/transact_failed
ok 10 /ipc/transact_status
ok 11 /ipc/transact_custom
ok 12 /ipc/transact_custom2
ok 13 /ipc/transact_custom3
ok 14 /ipc/transact_cancel
ok 15 /ipc/transact_cancel2
**
ERROR:../common/test_main.c:52:test_timeout_expired: assertion failed: (!"TIMEOUT")
Bail out! ERROR:../common/test_main.c:52:test_timeout_expired: assertion failed: (!"TIMEOUT")
make[1]: *** [../common/Makefile:141: test] Aborted
```

**Comment by monich** [comment]
That's a more interesting one. Please do this:
```
$ make -C unit/unit_ipc debug && unit/unit_ipc/build/debug/unit_ipc -v -p=/ipc/transact_2way
```
and if it fails, paste the output somewhere. `/ipc/transact_2way` is the test which seems  to be failing.

**Comment by monich** [comment]
And before you do that ☝️, check if https://github.com/mer-hybris/libgbinder/pull/72 helps.

**Comment by PureTryOut** [comment]
Both fail sadly.

```
# Start of ipc tests
[gbinder] Using aidl protocol for /dev/binder
[test_binder] Created /dev/binder (6) <=> /dev/binder-private (7) binder
[gbinder] Opened /dev/binder version 8
[gbinder] Using aidl protocol for /dev/binder
[gbinder] gbinder_ipc_register_local_object() 0x7f10cdd46a60 binder
[gbinder] Starting looper binder#1
[gbinder] < BC_ENTER_LOOPER
[gbinder] < 0c 63 00 00                                         .c..
[gbinder] gbinder_driver_write(8) 0/4
[test_binder] Thread 2663 is looper #1
[gbinder] gbinder_driver_write(8) 4/4 err 0
[gbinder] Looper binder#1 running
[gbinder] gbinder_driver_write_read(8) write 0/0 read 0/128
[gbinder] < BC_TRANSACTION 0x00000000 0x00000001
[gbinder]   00 00 40 00 04 00 00 00  74 00 65 00 73 00 74 00    ..@..... t.e.s.t.
[gbinder]   00 00 00 00 6d 65 73 73  61 67 65 00                ....mess age.
[gbinder] < 00 63 40 40 00 00 00 00  00 00 00 00 00 00 00 00    .c@@.... ........
[gbinder]   00 00 00 00 01 00 00 00  10 00 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00 1c 00 00 00  00 00 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00 00 9f b6 cd  10 7f 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00                                         ....
[gbinder] gbinder_driver_write_read(8) write 0/68 read 0/128
[test_binder] Stashed cmd 0x00007206 not for 2664
[gbinder] gbinder_driver_write_read(8) write 68/68 read 72/128 err 0
[gbinder] > 02 72 40 80 60 6a d4 cd  10 7f 00 00 00 00 00 00    .r@.`j.. ........
[gbinder]   00 00 00 00 02 00 00 00  00 00 00 00 66 0a 00 00    ........ ....f...
[gbinder]   e8 03 00 00 1c 00 00 00  00 00 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00 90 1b b7 cd  10 7f 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00 0c 72 00 00                             .....r..
[gbinder] > BR_TRANSACTION 0x7f10cdd46a60 (28 bytes)
[gbinder]   00 00 40 00 04 00 00 00  74 00 65 00 73 00 74 00    ..@..... t.e.s.t.
[gbinder]   00 00 00 00 6d 65 73 73  61 67 65 00                ....mess age.
[unit_ipc] test_transact_2way_incoming_proc() "test" 2
[gbinder] < BC_REPLY
[gbinder] < 01 63 40 40 00 00 00 00  00 00 00 00 00 00 00 00    .c@@.... ........
[gbinder]   00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00                                         ....
[gbinder] gbinder_driver_write(8) 0/68
[gbinder] gbinder_driver_write(8) 68/68 err 0
[gbinder] gbinder_driver_write_read(8) write 0/0 read 4/60
[test_binder] Stashed cmd 0x80407203 not for 2664
[gbinder] gbinder_driver_write_read(8) write 0/0 read 12/60 err 0
[gbinder] > 06 72 00 00 0c 72 00 00                             .r...r..
[gbinder] > BR_NOOP
[gbinder] > BR_TRANSACTION_COMPLETE
[gbinder] > BR_NOOP
[gbinder] gbinder_driver_write_read(8) write 68/68 read 0/128
[gbinder] gbinder_driver_write_read(8) write 68/68 read 76/128 err 0
[gbinder] > 03 72 40 80 00 00 00 00  00 00 00 00 00 00 00 00    .r@..... ........
[gbinder]   00 00 00 00 01 00 00 00  00 00 00 00 66 0a 00 00    ........ ....f...
[gbinder]   e8 03 00 00 0c 00 00 00  00 00 00 00 00 00 00 00    ........ ........
[gbinder]   00 00 00 00 40 9c b6 cd  10 7f 00 00 00 00 00 00    ....@... ........
[gbinder]   00 00 00 00 0c 72 00 00  0c 72 00 00                .....r.. .r..
[gbinder] > BR_REPLY (12 bytes)
[gbinder]   03 00 00 00 66 00 6f 00  6f 00 00 00                ....f.o. o...
[gbinder] > BR_NOOP
[gbinder] > BR_NOOP
[gbinder] < BC_FREE_BUFFER 0x7f10cdb71b90
[gbinder] < 03 63 08 40 90 1b b7 cd  10 7f 00 00                .c.@.... ....
[gbinder] gbinder_driver_write(8) 0/12
[gbinder] gbinder_driver_write(8) 12/12 err 0
[unit_ipc] test_transact_ok_done() 
[gbinder] < BC_FREE_BUFFER 0x7f10cdb69c40
[gbinder] < 03 63 08 40 40 9c b6 cd  10 7f 00 00                .c.@@... ....
[gbinder] gbinder_driver_write(8) 0/12
[gbinder] gbinder_driver_write(8) 12/12 err 0
[unit_ipc] waiting for GBinderIpc to get destroyed
[gbinder] gbinder_ipc_local_object_disposed() 0x7f10cdd46a60 binder
[gbinder] gbinder_driver_write_read(8) write 0/0 read 0/128 err 0
[gbinder] gbinder_ipc_dispose() /dev/binder
[gbinder] Stopping looper binder#1
[gbinder] Looper binder#1 is requested to exit
[gbinder] < BC_EXIT_LOOPER
[gbinder] < 0d 63 00 00                                         .c..
[gbinder] gbinder_driver_write(8) 0/4
[test_binder] Thread 2663 is no longer a looper
[gbinder] gbinder_driver_write(8) 4/4 err 0
[gbinder] Looper binder#1 is abandoned
[test_binder] Waiting for loopers to exit...
**
ERROR:../common/test_main.c:52:test_timeout_expired: assertion failed: (!"TIMEOUT")
Bail out! ERROR:../common/test_main.c:52:test_timeout_expired: assertion failed: (!"TIMEOUT")
Aborted
```

**Comment by PureTryOut** [comment]
Consistently, I haven't had it pass once yet.

**Comment by monich** [comment]
Try https://github.com/mer-hybris/libgbinder/pull/74 then. With those patches applied, all tests are passing for me a musl based linux.

**Comment by monich** [comment]
The issue should be resolved in version 1.1.11
